js/index.js

```diff
@@ -1,33 +1,53 @@
 // ITERATION 1
-
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+  const price = product.querySelector('.price span')
+  const quantity = product.querySelector('.quantity input')
+
+  // price.innerHTML
+  // quantity.value
+
+  const totalPrice = price.innerHTML * quantity.value
+
+  const subtotal = product.querySelector('.subtotal span')
+
+  subtotal.innerText = totalPrice
+
+  return totalPrice
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  const allProducts = document.querySelectorAll('.product');
+  for (let i = 0; i < allProducts.length; i++) {
+    updateSubtotal(allProducts[i])
+  }
+  let valueSum = 0
+  for (let i = 0; i < allProducts.length; i++) {
+    valueSum += updateSubtotal(allProducts[i])
+  }
+  const totalValue = document.querySelector('#total-value span')
+
+  totalValue.innerText = valueSum
+}
 
-  // ITERATION 2
-  //... your code goes here
 
-  // ITERATION 3
-  //... your code goes here
-}
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+
+  // const buttons = document.querySelectorAll('#btn btn-remove')
+  // for (let i = 0; i < buttons.length; i++) {
+  //   buttons[i].onclick = function (event) {
+  //     event.currentTarget = quantity.remove
+
+  target.parentNode.parentNode.remove();
```

Comment:
Te pongo por aquí una posible solución:
function removeProduct(event) {
  const target = event.currentTarget;
  console.log('The target in remove is:', target);
 
    let padre = document.querySelector('tbody')
    const hijo = target.parentNode.parentNode
    let removeElement = padre.removeChild(hijo)
    
    calculateAll()
}
El método remove.Child coge el padre y en el paréntesis le decimos el hijo a borrar.